### DISCUSSION TITLE
Generate raw sql independantly of models

### DISCUSSION THREAD

**Discussion by vianmora** [sqlalchemy/sqlalchemy#11052] [Ideas]
Hello :)  

perhaps it's already possible, but i didn't find it :/ 
I would like to have a python function to generate postgres sql without to connect to the engine

an example, would help perhaps, something like this : 

```
from sqlachemy import postgres_orm

sql_rename_table = postgres_orm.get_sql_rename_table(sql_table_target = "my_schema.my_table", new_name="my_table_renamed")

print(sql_rename_table) = "ALTER TABLE my_schema.my_table RENAME TO "my_table_renamed";
```

I've already implemented, for a mini-project, some queries like this one, and i wanted to store them somewhere it could be usefull for other people :)  

- add_column
- add_foreign_key
- add_index
- ...

I'm  somehow a beguinner in contributing in open-source, what could it be the process to propose my code ? And where do you think it's the best  to place it ? 

Thank you for you amazing tool by the way !!

**Comment by zzzeek** [comment]
we have this!  for DDL like RENAME TABLE, it's in a project called [alembic](https://alembic.sqlalchemy.org/).   

```
>>> from alembic.ddl.base import RenameTable
>>> RenameTable("x", "y")
<alembic.ddl.base.RenameTable object at 0x7fbc22456c00>
>>> print(RenameTable("x", "y"))
ALTER TABLE x RENAME TO y
>>> from sqlalchemy.dialects import postgresql
>>> print(RenameTable("x", "y").compile(dialect=postgresql.dialect()))
ALTER TABLE x RENAME TO y

```

general background on stringifying SQLAlchemy elements is at https://docs.sqlalchemy.org/en/20/faq/sqlexpressions.html#how-do-i-render-sql-expressions-as-strings-possibly-with-bound-parameters-inlined
